programs/bpf_loader/src/lib.rs

```diff
@@ -103,94 +103,96 @@ pub fn load_program_from_bytes(
     Ok(loaded_program)
 }
 
```

Comment:
This used to be a function `direct_deploy_program()` invoking an inner macro `deploy_program!()`.
And now is a macro `deploy_program!()` invoking an inner function `deploy_program_internal()`.
Swapping the roles of the macro and wrapper function allows us to get rid of some lifetime shenanigans and explicit `drop()`. It is a pure syntactic refactoring and should have no effect on the behavior.